index.js

```diff
@@ -1,19 +1,69 @@
 /* globals require, __dirname, process */
 
 const path = require('path');
-const fs = require('fs');
+const { promises: fs } = require('fs');
 const builder = require("adguard-hostlists-builder");
+const SVGParser = require('svg-parser');
+
+// build services.json from services folder
+(async() => {
+    try {
+        // variable to store json's objects
+        let servicesData = []
+        // array with all service names
+        const dirNames = await fs.readdir('./services');
+        // collect data from service files
+        const collectData = dirNames.map(async(dirName) => {
+            // get file path for reading data
```

Comment:
Move `async(dirName) => { ....` to a separate function (you can call it`readServiceData` for instance).

Do SVG validation inside that function.